net/libfab/libfab.c

```diff
@@ -0,0 +1,2829 @@
+/* -*- C -*- */
+/*
+ * Copyright (c) 2020 Seagate Technology LLC and/or its Affiliates
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * For any questions about this software or licensing,
+ * please email opensource@seagate.com or cortx-questions@seagate.com.
+ *
+ */
+
+/**
+ * @addtogroup netlibfab
+ *
+ * @{
+ */
+
+#define M0_TRACE_SUBSYSTEM M0_TRACE_SUBSYS_NET
+#include "lib/trace.h"          /* M0_ENTRY() */
+#include <netinet/in.h>         /* INET_ADDRSTRLEN */
+#include <arpa/inet.h>          /* inet_pton, htons */
+#include <stdlib.h>             /* atoi */
+#include <sys/epoll.h>          /* struct epoll_event */
+#include <unistd.h>             /* close */
+#include "net/buffer_pool.h"    /* struct m0_net_buffer_pool */
+#include "net/net.h"            /* struct m0_net_domain */
+#include "lib/errno.h"          /* errno */
+#include "lib/memory.h"         /* M0_ALLOC_PTR()*/
+#include "libfab_internal.h"
+#include "net/net_internal.h"   /* m0_net__buffer_invariant() */
+
+static char     *providers[] = { "verbs", "tcp", "sockets" };
+static char     *portf[]  = { "unix", "inet", "inet6" };
+static char     *socktype[] = { "stream", "dgram" };
+static uint64_t  mr_key_idx = 0;
+/* Bitmap of used transfer machine identifiers */
+static char      fab_autotm[1024] = {}; 
```

Comment:
```
/* 
 * Bitmap of used transfer machine identifiers. 1 is for used, and 0 is for free.
 */
static uint8_t fab_autotm[1024] = {};
```

But also, we have m0_bitmap.
Can we use that?

Reply:
Used uint8_t instead of char.